Use Backbonejs' router to switch between different "page" views

bgw opened the pull request:
**I don't think this should be merged yet**, but I wanted to open the pull request now to allow for future discussion. I'm sure there will be a lot of questions about the "right way" to do this.

kalgynirae commented:
This all looks pretty good to me. So, what do you think the next steps are? I can probably work on some stuff this evening, assuming the OS homework doesn't take too long.

bgw commented:
> So, what do you think the next steps are?
- Make sure we have a view for each page (even if it is just a stub)
  - We should use that template we have for the login page
- ~~Make sure each view can be destroyed properly (look at [`Backbone.View.remove()`](http://backbonejs.org/#View-remove))~~
- Some of our models have callbacks that would prevent them from being garbage collected. Write some function for each of them that we can use to destroy them.
  - ~~[`Backbone.Model.destroy()`](http://backbonejs.org/#Model-destroy) is already defined but has a different meaning than what we want.~~ _We should define a `close` function (see my comment below)._
- ~~Load the `gps.Tracker` module when the application first loads and try to start tracking immediately (gives the device as much time as possible to get a GPS fix)~~
  - ~~If we were denied permission by the user, try to start tracking again when they hit the start button~~
- ~~Figure out some way to hide the navigation bar on the login screen~~
- Finally, figure out some way of having multiple page views loaded simultaneously, being able to scroll horizontally between them.
  - I feel like pressing a button should have a css sliding transition too to make this gesture clearer to the user.

bgw commented:
[There's some discussion on StackOverflow about the "right way" to dispose of views and models](http://stackoverflow.com/questions/7379263/disposing-of-view-and-model-objects-in-backbone-js). The example given there (reprinted) is:

``` .javascript
MyView = Backbone.View.extend({
    initialize: function() {
        this.model.bind("change", this.modelChanged, this);
    },

    modelChanged: function() {
        // ... do stuff here
    },

    close: function() {
        this.remove();
        this.unbind();
        this.model.unbind("change", this.modelChanged);
    }
});
```

Essentially, the suggestion is to have a `close` function in all custom views and models that disposes of things properly.

Commit:
Remove unnecessary usage of `_.bind`

Commit:
Increase requirejs timeout period

The build times out on my netbook otherwise

Commit:
Add code to dispose properly of gps view

Switching tabs will now dispose of and recreate the gps "track" view as
expected.

Commit:
Merge branch 'features/router' of github.com:routemaster/routemaster-frontend into features/router

Conflicts:
	src/js/urlHandler.js

kalgynirae committed:
Add list views; add dummy history view

* Fix the attribution on the Leaflet map
* Fix 'listing' style so it applies to ol and ul

Commit:
Fix routing on initial page load

The url fragment would be ignored on initial page load. This is no
longer the case. `Backbone.history.start()` has to be called **after**
the `Router` is constructed for it to receive the routing request.

Commit:
Hide the "Powered by Leaflet" attribution

It's not required and would take up a lot of screen estate on a mobile
device. Plus, the link could be clicked accidentally by a user.

Commit:
Change the way tracking is handled in gps model

- The GPS is activated at application start
- Instead of calling `startTracking` and `stopTracking` in the model,
  you should now use `model.set("tracking", true)`. (looser bindings)

Commit:
Improve css styling for list views

It's not perfect, but the list items now look a bit more "clickable".

Commit:
Improve button styling and display

The "start" and "stop" buttons on the tracking page have been simplified
and enlarged. Furthermore, only one of the two buttons shows at a time
now.

kalgynirae committed:
Remove footer

Commit:
Fix leaflet display bug

Leaflet has a bug where, if placed in a dynamically created div in a
certain way, the size of the display is miscalcuated. This forces the
map to recompute the size.

kalgynirae committed:
Make list items expand when clicked
